### ISSUE TITLE
Credentials from authorization code and tokenEndpoint

### ISSUE THREAD

**Issue by stulov** [dart-lang/oauth2#132]
Hello!

I'm implementing Spotify Authorization Code Flow with PKCE using their native spotify-auth library.
Once I receive the authorization code as a result of the native LoginActivity I'd like to exchange the code for the access token and the refresh token by requesting my tokenEndpoint.

```dart
final code = await _spotifyAuth.getAuthorizationCode(); // gets the code from native login activity via method channel
```

It seems that, for now, in order to create a OAuth Client I have to manually request the tokenEndpoint using some http client, get access token and refresh token and then build Credentials and pass them into Client. Something like this:

```dart
final code = await _spotifyAuth.getAuthorizationCode();

final res = await _dio.post<Map<String, dynamic>>(
  _tokenEndpoint,
  data: {
    'grant_type': 'authorization_code',
    'code': code,
);

final credentials = oauth2.Credentials(
  res.data['access_token'],
  refreshToken: res.data['refresh_token'],
  tokenEndpoint: _tokenEndpoint,
);

final client = oauth2.Client(credentials, identifier: _spotifyAuth.clientId);
```

So, basically I'm wondering if it is possible to omit the step where I'm requesting credentials by hand and instantiate Client with authorization code and tokenEndpoint at once?
